source/geometries/BlackBox.cc

```diff
@@ -0,0 +1,238 @@
+                            "Hole Diameter");
+    hole_diameter_cmd.SetParameterName("hole_diameter", true);
+    hole_diameter_cmd.SetUnitCategory("Length");
+
+    G4GenericMessenger::Command&  hole_x_cmd =
+      msg_->DeclareProperty("hole_x", hole_x_,
+                            "Hole X");
+    hole_x_cmd.SetParameterName("hole_x", true);
+    hole_x_cmd.SetUnitCategory("Length");
+
+    G4GenericMessenger::Command&  hole_y_cmd =
+      msg_->DeclareProperty("hole_y", hole_y_,
+                            "Hole Y");
+    hole_y_cmd.SetParameterName("hole_y", true);
+    hole_y_cmd.SetUnitCategory("Length");
+  }
+
+  BlackBox::~BlackBox()
+  {
+    delete msg_;
+  }
+
+  void BlackBox::Construct()
+  {
+
+  dice_ = new BlackBoxSiPMBoard();
+
+  // WORLD /////////////////////////////////////////////////
+
+  G4String world_name = "WORLD";
+
+  G4Material* world_mat = G4NistManager::Instance()->FindOrBuildMaterial("G4_AIR");
+
+  world_mat->SetMaterialPropertiesTable(opticalprops::Vacuum());
+
+  G4Box* world_solid_vol =
+    new G4Box(world_name, world_xy_/2., world_xy_/2., world_z_/2.);
+
+  G4LogicalVolume* world_logic_vol =
+    new G4LogicalVolume(world_solid_vol, world_mat, world_name);
+  world_logic_vol->SetVisAttributes(G4VisAttributes::GetInvisible());
+  GeometryBase::SetLogicalVolume(world_logic_vol);
+
+  // BLACK BOX /////////////////////////////////////////////////
+
+  G4String box_name = "BLACK BOX";
+
+  G4Box* box_solid_vol =
+    new G4Box(box_name, box_xy_/2., box_xy_/2., box_z_/2.);
+
+  G4LogicalVolume* box_logic_vol =
+    new G4LogicalVolume(box_solid_vol, world_mat, box_name);
+  box_logic_vol->SetVisAttributes(G4VisAttributes::GetInvisible());
+
+  G4VPhysicalVolume* box_phys_vol =
+   new G4PVPlacement(0, G4ThreeVector(0.,0.,0.),
+                     box_logic_vol, box_name, world_logic_vol,
+                     false, 0, false);
+
+  // DICE BOARD //////////////////////////////////////////////
+
+  dice_->SetMotherPhysicalVolume(box_phys_vol);
+  dice_->SetMaskThickness    (mask_thickn_);
+  dice_->SetHoleDiameter     (hole_diameter_);
+  dice_->SetHoleX            (hole_x_);
+  dice_->SetHoleY            (hole_y_);
+  dice_->Construct();
+  G4LogicalVolume* dice_board_logic = dice_->GetLogicalVolume();
+
+  ////Dice Board placement
+  G4RotationMatrix* rot = new G4RotationMatrix();
+  rot -> rotateY(rotation_);
+  new G4PVPlacement(rot, dice_board_pos_, dice_board_logic,
+            	      dice_board_logic->GetName(), box_logic_vol, false, 1, false);
+
+  // SAPPHIRE //////////////////////////////////////////////////
+
+  if (sapphire_) {
+
+    G4String sapphire_name = "SAPPHIRE";
+
+    G4Material* sapphire_mat = materials::Sapphire();
+
+    sapphire_mat->SetMaterialPropertiesTable(opticalprops::Sapphire());
+
+    G4double pedot_thickn  = 150. *nm;
+    G4double sapphire_thickn  = 6. *mm;
+    G4double sapphire_diam = 20 *cm;
+    G4double window_thickn  = sapphire_thickn;
```

Comment:
I think that defining `window_thickn`  is not necessary. You can simply do 

`G4double window_thickn = 6. * mm;`

and then modify it if PEDOT is added.

Reply:
Done!